cmake/compile_definitions/windows.cmake

```diff
@@ -5,6 +5,7 @@ add_compile_definitions(SUNSHINE_PLATFORM="windows")
 enable_language(RC)
 set(CMAKE_RC_COMPILER windres)
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static")
+list(APPEND SUNSHINE_COMPILE_OPTIONS -flarge-source-files)
```

Comment:
Is this truly necessary?

Reply:
The compiler spits out a lot of warnings about misleading indentation and large source files when compiling anything that includes the WinRT headers. This silences those. I suppose the flag is optional, but I added it in the interest of minimizing compiler warnings.

Reply:
Okay... is it possible to enable this flag only for those headers?

Reply:
It would need to be called out explicitly for every source file that includes display.h or a WinRT header in the windows build.
It's doable for now,

```
set_source_files_properties("${CMAKE_SOURCE_DIR}/src/platform/windows/display_wgc.cpp"
        PROPERTIES COMPILE_FLAGS "-flarge-source-files")
```

but I think this could get out of hand quickly over time so I'm less sure it's a good idea.
Will proceed if you insist.

Reply:
We're actually running into this warning already. See https://github.com/LizardByte/Sunshine/pull/2165#issuecomment-1962004615

I'd like to resolve it a different way if possible. It was pointed out to me that these files are probably not being treated as system files, although they should be.

Reply:
I kinda got nothing other than adding this option. I see in the other pull request you disabled the misleading indentation warning but frankly I feel that's more dangerous than adding this option.

>I'd like to resolve it a different way if possible.

I'd be curious to hear what pitfalls you've experienced with `-flarge-source-files`!

Reply:
> note: adding '-flarge-source-files' will allow for more column-tracking support, at the expense of compilation time and memory

I'm not really wanting to increase CI time or memory consumption.

We lint our source files, so it's pretty unlikely the misleading indentation would come from our code.

Reply:
There doesn't appear to be any misleading indentation after all. In fact, I found the "warning" occuring in a Boost header while compiling sunshine. I think the warning is actually a notification that, "Here's where `gcc` stopped counting columns". I'll ignore it for now.

(I only noticed now because `note`s and `warning`s are different colors in the msys terminal)